packages/middleware-sdk-ec2/src/index.ts

```diff
@@ -1,40 +1,35 @@
+import { EC2ClientResolvedConfig } from "@aws-sdk/client-ec2";
 import { HttpRequest } from "@aws-sdk/protocol-http";
 import { SignatureV4 } from "@aws-sdk/signature-v4";
 import {
-  Credentials,
-  Endpoint,
-  HashConstructor,
   InitializeHandler,
   InitializeHandlerArguments,
   InitializeHandlerOptions,
   InitializeHandlerOutput,
   InitializeMiddleware,
-  MemoizedProvider,
   MetadataBearer,
   Pluggable,
-  Provider,
 } from "@aws-sdk/types";
 import { formatUrl } from "@aws-sdk/util-format-url";
 
-interface PreviouslyResolved {
-  credentials: MemoizedProvider<Credentials>;
-  endpoint: Provider<Endpoint>;
-  region: Provider<string>;
-  sha256: HashConstructor;
-  signingEscapePath: boolean;
-}
-
 const version = "2016-11-15";
 
 //an initialize middleware to add PresignUrl to input
-export function copySnapshotPresignedUrlMiddleware(options: PreviouslyResolved): InitializeMiddleware<any, any> {
+export function copySnapshotPresignedUrlMiddleware(options: EC2ClientResolvedConfig): InitializeMiddleware<any, any> {
```

Comment:
Even though TypeScript doesn't care about the circular dependencies, I prefer using `PreviouslyResolved` type so it's less coupled.

Reply:
ok, I'll do that